load-tests/class-scenario.js

```diff
@@ -0,0 +1,244 @@
+import http from "k6/http";
+import exec from "k6/execution";
+import { Trend } from "k6/metrics";
+
+import { renkuLogin } from "./oauth.js";
+import { check, fail, sleep } from "k6";
+
+import {
+  baseUrl,
+  credentials,
+  sampleGitProjectUrl,
+  serverOptions,
+} from "./config.js";
+
+export const options = {
+  scenarios: {
+    lecture: {
+      executor: "per-vu-iterations",
+      vus: 10, // tested up to 30
+      iterations: 1,
+    },
+  },
+};
+
+// k6 custom metrics
+const sessionStartupTrend = new Trend("session_startup");
+
+function httpRetry(httpRequest, n, logMessage) {
```

Comment:
@li-il-li sorry for the long delay in reviewing this. But I think that this function hides exactly the problems that we want the load tests to uncover.

Can you please remove it and I will approve.

If requests need to be retried in the load tests then please submit issues for it in the corresponding components.